src/test/java/uk/gov/hmcts/reform/civil/handler/event/BundleCreationTriggerEventHandlerTest.java

```diff
@@ -0,0 +1,151 @@
+package uk.gov.hmcts.reform.civil.handler.event;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.ccd.client.model.StartEventResponse;
+import uk.gov.hmcts.reform.civil.enums.YesOrNo;
+import uk.gov.hmcts.reform.civil.event.BundleCreationTriggerEvent;
+import uk.gov.hmcts.reform.civil.helpers.CaseDetailsConverter;
+import uk.gov.hmcts.reform.civil.model.CaseData;
+import uk.gov.hmcts.reform.civil.model.Party;
+import uk.gov.hmcts.reform.civil.model.ServedDocumentFiles;
+import uk.gov.hmcts.reform.civil.model.bundle.Bundle;
+import uk.gov.hmcts.reform.civil.model.bundle.BundleCreateResponse;
+import uk.gov.hmcts.reform.civil.model.bundle.BundleData;
+import uk.gov.hmcts.reform.civil.model.bundle.BundleDetails;
+import uk.gov.hmcts.reform.civil.model.caseprogression.UploadEvidenceDocumentType;
+import uk.gov.hmcts.reform.civil.model.caseprogression.UploadEvidenceExpert;
+import uk.gov.hmcts.reform.civil.model.caseprogression.UploadEvidenceWitness;
+import uk.gov.hmcts.reform.civil.model.common.DynamicList;
+import uk.gov.hmcts.reform.civil.model.common.DynamicListElement;
+import uk.gov.hmcts.reform.civil.model.common.Element;
+import uk.gov.hmcts.reform.civil.model.documents.CaseDocument;
+import uk.gov.hmcts.reform.civil.model.documents.Document;
+import uk.gov.hmcts.reform.civil.model.documents.DocumentType;
+import uk.gov.hmcts.reform.civil.sampledata.CaseDetailsBuilder;
+import uk.gov.hmcts.reform.civil.service.CoreCaseDataService;
+import uk.gov.hmcts.reform.civil.service.bundle.BundleCreationService;
+import uk.gov.hmcts.reform.civil.utils.ElementUtils;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.mockito.ArgumentMatchers.anyMap;
+import static org.mockito.Mockito.when;
+import static uk.gov.hmcts.reform.civil.callback.CaseEvent.CREATE_BUNDLE;
+
+@ExtendWith(SpringExtension.class)
+public class BundleCreationTriggerEventHandlerTest {
+
+    @Mock
+    private BundleCreationService bundleCreationService;
+    @Mock
+    CoreCaseDataService coreCaseDataService;
+    @Mock
+    CaseDetailsConverter caseDetailsConverter;
+    private BundleCreationTriggerEvent event = new BundleCreationTriggerEvent(1L);
+    private BundleCreateResponse bundleCreateResponse;
+    @InjectMocks
+    private BundleCreationTriggerEventHandler bundleCreationTriggerEventHandler;
+    private final String testUrl = "url";
+    private final String testFileName = "testFileName.pdf";
+    CaseData caseData;
+    CaseDetails caseDetails;
+
+    @BeforeEach
+    public void setup() {
```

Comment:
That's a very long setup method. Try to split it in smaller methods stating clearly what bits they are setting up. It would improve readability.

Reply:
done